django/db/models/sql/query.py

```diff
@@ -551,7 +555,14 @@ def explain(self, using, format=None, **options):
         q.explain_format = format
         q.explain_options = options
         compiler = q.get_compiler(using=using)
-        return '\n'.join(compiler.explain_query())
+        if connection.vendor == 'postgresql':
+            if format == 'json':
+                return json.loads(json.dumps(ast.literal_eval('\n'.join(compiler.explain_query()))))[0]
```

Comment:
I don't think the `ast.literal_eval` approach is the best. I only included the code as an example on the ticket. It would be better to never force the result into a string in the first place.